docs/source/json-api/index.rst

```diff
@@ -165,34 +136,58 @@ token.  The default "header" is fine.  Under "Payload", fill in:
+For the Sandbox this corresponds to the ``--ledgerid MyLedger`` flag.
+
+.. note:: The value of ``applicationId`` will be used for commands submitted using that token.
+
+The value for ``actAs`` is specified as a list and you provide it with the party that you want to use.
+Such as the example which uses ``Alice`` for a party. Each request can only be for one party.
+For example you couldn't have ``actAs`` defined as ``["Alice", "Bob"]``.
 
-Under "Verify Signature", put ``secret`` as the secret (*not* base64
-encoded); that is the hardcoded secret for testing.
+For the Sandbox any string will create use a party with that name if it exists, 
+or create it if it does not. This is known as "implicit party allocation" and is unique to the Sandbox.
+
+For non-Sandbox ledgers the party must already exist on the ledger in order to be used.
+This is known as "explicit party allocation" and is common on all non-Sandbox ledgers.
+
+.. note:: To explicitly allocate parties you can use :doc:`DAML Script </daml-script/index>` or ``daml ledger allocate-parties``
+
+Under "Verify Signature", put ``secret`` (or any other text) as the secret (*not* base64 encoded).
+
+.. note:: At the moment ``secret`` is not validated
 
 Then the "Encoded" box should have your **token**, ready for passing to
 the service as described in the following sections.
 
 Alternatively, here are two tokens you can use for testing:
 
-- ``{"https://daml.com/ledger-api": {"ledgerId": "MyLedger", "applicationId": "foobar", "actAs": ["Alice"]}}``
-  ``eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJBbGljZSJdfX0.VdDI96mw5hrfM5ZNxLyetSVwcD7XtLT4dIdHIOa9lcU``
+``{"https://daml.com/ledger-api": {"ledgerId": "MyLedger", "applicationId": "HTTP-JSON-API-Gateway", "actAs": ["Alice"]}}``:
 
-- ``{"https://daml.com/ledger-api": {"ledgerId": "MyLedger", "applicationId": "foobar", "actAs": ["Bob"]}}``
-  ``eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJCb2IiXX19.zU-iMSFG90na8IHacrS25xho3u6AKnSlTKbvpkaSyYw``
+.. code-block:: none
 
-For production use, we have a tool in development for generating proper
-RSA-encrypted tokens locally, which will arrive when the service also
-supports such tokens.
+    eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJIVFRQLUpTT04tQVBJLUdhdGV3YXkiLCJhY3RBcyI6WyJBbGljZSJdfX0.34zzF_fbWv7p60r5s1kKzwndvGdsJDX-W4Xhm4oVdpk
 
-Passing token with HTTP
-=======================
 
-Set HTTP header ``Authorization: Bearer copy-paste-token-here`` for
-normal requests.
+``{"https://daml.com/ledger-api": {"ledgerId": "MyLedger", "applicationId": "HTTP-JSON-API-Gateway", "actAs": ["Bob"]}}``:
 
-Passing token with WebSockets
-=============================
+.. code-block:: none
+
```

Comment:
Does this actually render as code? I thought two newlines mean it no longer belongs to the code block.

Reply:
Yes all is rendering correctly, there's 1 newline after the `code-block` delegation and two newlines separating the inline code block that displays the json-encoded token.